contracts/utils/convex/ConvexStakingWrapperYield.sol

```diff
@@ -0,0 +1,111 @@
+//https://etherscan.io/address/0x3ba207c25a278524e1cc7faaea950753049072a4#code
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+pragma experimental ABIEncoderV2;
+
+import './ConvexStakingWrapper.sol';
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+
+struct Balances {
+    uint128 art; // Debt amount
+    uint128 ink; // Collateral amount
+}
+
+interface ICauldron {
+    /// @dev Each vault records debt and collateral balances_.
+    function balances(bytes12 vault) external view returns (Balances memory);
+}
+
+//Staking wrapper for Yield platform
+//use convex LP positions as collateral while still receiving rewards
+contract ConvexStakingWrapperYield is ConvexStakingWrapper,AccessControl {
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+
+    mapping(address=>bytes12[]) public vaults;
+    address cauldron;
+
+    constructor() public {}
+
+    function initialize(
+        address _curveToken,
+        address _convexToken,
+        address _convexPool,
+        uint256 _poolId,
+        address _vault
+    ) external override auth{
+        require(!isInit, 'already init');
+        owner = address(0xa3C5A1e09150B75ff251c1a7815A07182c3de2FB); //TODO: Find why this needs to be set to convex multisig
+        emit OwnershipTransferred(address(0), owner);//TODO: Find why this needs to be done
```

Comment:
This is done so that there is an event trail for ownership changes, including the first one